src/Razor/src/Microsoft.VisualStudio.RazorExtension/Microsoft.VisualStudio.RazorExtension.csproj

```diff
@@ -136,6 +141,21 @@
     </ItemGroup>
   </Target>
 
+  <Target Name="_BuildRazorGrammar">
+    <MSBuild Projects="..\Microsoft.AspNetCore.Razor.VSCode.Extension\Microsoft.AspNetCore.Razor.VSCode.Extension.npmproj"
+        Targets="Build" />
+  </Target>
+
+  <Target Name="_IncludeRazorGrammar" DependsOnTargets="PrepareForBuild;_BuildRazorGrammar"  BeforeTargets="CoreCompile">
+    <ItemGroup>
+      <Content Include="..\Microsoft.AspNetCore.Razor.VSCode.Extension\syntaxes\aspnetcorerazor.tmLanguage.json">
```

Comment:
Should the tmLanguage.json move to somewhere common now that it is used in VS as well?

Reply:
Ya it's a tough problem because we need node to compile the grammar. We could technically add a separate Razor.Grammar npmproj and then rewire the VSCode.extensions and VSIX project to pull/build it but I'm not sure how much value that would add.

If you have any alternative ideas I'm all ears 😁